contracts/protocol/facets/MetaTransactionsHandlerFacet.sol

```diff
@@ -252,8 +252,17 @@ contract MetaTransactionsHandlerFacet is IBosonMetaTransactionsHandler, Protocol
         (bool success, bytes memory returnData) = address(this).call{ value: msg.value }(_functionSignature);
 
         // If error, return error message
-        string memory errorMessage = (returnData.length == 0) ? FUNCTION_CALL_NOT_SUCCESSFUL : (string(returnData));
-        require(success, errorMessage);
+        if (!success) {
+            if (returnData.length > 0) {
+                // bubble up the error
+                assembly {
+                    revert(add(32, returnData), mload(returnData))
+                }
+            } else {
+                // Reverts with default message
+                revert(FUNCTION_CALL_NOT_SUCCESSFUL);
+            }
+        }
```

Comment:
this seems like a step backwards, I wonder why this is like this ?

Reply:
`returnData` contains some bytes at the beginning, which are not the actual revert reason.

When it's cast to string (`string(returnData)`) solidity does not really care that the bytes are not valid string bytes. So if that's then passed as a revert message, the client cannot decode it into a valid string.

For some reason, this wasn't a problem in the past (probably hardhat/ethers ignored the invalid part of the string), but now it is. 

In any case, this is now the correct implementation for bubbling up the error, i.e. the error is exactly the same as the one, raised by the contract that reverted.

Reply:
And without this code change, some tests fail.